include/ofi_atom.h

```diff
@@ -249,6 +275,26 @@ typedef atomic_long	ofi_atomic_int64_t;
 					     int##radix##_t desired)					\
 	{												\
 		return ofi_atomic_cas_bool##radix(atomic, expected, desired);				\
+	}												\
+	static inline											\
+	bool ofi_atomic_compare_exchange_weak##radix(ofi_atomic##radix##_t *atomic,			\
+					int##radix##_t *expected,					\
+					int##radix##_t desired)						\
+	{												\
+		return __atomic_compare_exchange_n(ofi_atomic_ptr(atomic), expected, desired,		\
```

Comment:
This (and the other two) may not be available on all OSes. Please go back a few blocks to see how `atomic_add_and_fetch` is handled (with definitions in `osd.h`).

Reply:
@j-xiong, I'm unsure what the windows equivalent of these functions should be.

What I have is below. I think the compare exchange and store should work. But I don't know what the load should be? I can't find an interlockedAcquire() function. Any ideas?

```
1034 #define ofi_atomic_compare_exchange_weak(radix, ptr, expected, desired, \
1035 »·······»·······»·······»·······»······· succ_memmodel, fail_memmodel) \
1036 »·······InterlockedCompareExchange((ofi_atomic_int_##radix##_t volatile *)ptr, desired, expected)
1037 #define ofi_atomic_store_explicit(radix, ptr, value, memmodel) \
1038 »·······InterlockedExchange((ofi_atomic_int_##radix##_t volatile *)ptr, value)
1039 #define ofi_atomic_load_explicit(radix, ptr, memmodel) \
1040 »·······__atomic_load_n(ptr, memmodel)
```

Reply:
IIUC, this code is only used by shm. Right? Can we disable shm compilation on windows? I do not think shm works on windows any way.

Reply:
shm shouldn't be building on windows. It will only build if CMA is found which is only on linux
https://github.com/ofiwg/libfabric/blob/main/prov/shm/configure.m4#L89

Reply:
This file and the functions in it are built on windows and used there.

I'm not sure what the windows equivalent to load is, but guessing you could use Add(0) or just return the value.